compiler/rustc_privacy/src/lib.rs

```diff
@@ -684,75 +681,26 @@ impl<'tcx> Visitor<'tcx> for EmbargoVisitor<'tcx> {
                     &self.effective_visibilities,
                 );
 
-                self.update_eff_vis(item.owner_id.def_id, impl_ev, None, Level::Direct);
+                if let Some(impl_ev) = impl_ev {
+                    self.update_eff_vis(item.owner_id.def_id, impl_ev, None, Level::Direct);
+                }
                 impl_ev
             }
             _ => self.get(item.owner_id.def_id),
         };
 
-        // Update levels of nested things.
+        // Update levels of nested things and mark all items
+        // in interfaces of reachable items as reachable.
         match item.kind {
-            hir::ItemKind::Enum(ref def, _) => {
-                for variant in def.variants {
-                    self.update(variant.def_id, item_ev, Level::Reachable);
-                    let variant_ev = self.get(variant.def_id);
-                    if let Some(ctor_def_id) = variant.data.ctor_def_id() {
-                        self.update(ctor_def_id, variant_ev, Level::Reachable);
-                    }
-                    for field in variant.data.fields() {
-                        self.update(field.def_id, variant_ev, Level::Reachable);
-                    }
-                }
-            }
-            hir::ItemKind::Impl(ref impl_) => {
-                for impl_item_ref in impl_.items {
-                    self.update(impl_item_ref.id.owner_id.def_id, item_ev, Level::Direct);
-                }
-            }
-            hir::ItemKind::Trait(.., trait_item_refs) => {
-                for trait_item_ref in trait_item_refs {
-                    self.update(trait_item_ref.id.owner_id.def_id, item_ev, Level::Reachable);
-                }
-            }
-            hir::ItemKind::Struct(ref def, _) | hir::ItemKind::Union(ref def, _) => {
-                if let Some(ctor_def_id) = def.ctor_def_id() {
-                    self.update(ctor_def_id, item_ev, Level::Reachable);
-                }
-                for field in def.fields() {
-                    self.update(field.def_id, item_ev, Level::Reachable);
-                }
-            }
+            // The interface is empty, and no nested items.
+            hir::ItemKind::Use(..)
+            | hir::ItemKind::ExternCrate(..)
+            | hir::ItemKind::GlobalAsm(..) => {}
+            // The interface is empty, and all nested items are processed by `visit_item`.
+            hir::ItemKind::Mod(..) => {}
             hir::ItemKind::Macro(ref macro_def, _) => {
                 self.update_reachability_from_macro(item.owner_id.def_id, macro_def);
             }
-            hir::ItemKind::ForeignMod { items, .. } => {
-                for foreign_item in items {
-                    self.update(foreign_item.id.owner_id.def_id, item_ev, Level::Reachable);
```

Comment:
Why isn't this update needed any more?

Reply:
It wasn't needed previously either, both nominal and effective visibilities are meaningless for `extern { ... }` items themselves, and they are not added to the table - https://github.com/rust-lang/rust/blob/master/tests/ui/privacy/effective_visibilities.rs#L10